### ISSUE TITLE
Java 8 bug when two models return Map

### ISSUE THREAD

**Issue by shifu** [teamcarma/swagger-jaxrs-doclet#30]
```java
public class ModelA {
  public Map getData() {...}
}

public class ModelB {
  public Map getData() {...}
}
```

If I have an API that references both ModelA and B, it will generate the below error.  It works fine on Java 7.

```
[ERROR] javadoc: error - In doclet class com.carma.swagger.doclet.ServiceDoclet,  method start has thrown an exception java.lang.reflect.InvocationTargetException
[ERROR] java.lang.IllegalStateException: dupe models, method : 
...
[ERROR] at com.carma.swagger.doclet.parser.CrossClassApiParser.addApiModels(CrossClassApiParser.java:222)
[ERROR] at com.carma.swagger.doclet.parser.CrossClassApiParser.parse(CrossClassApiParser.java:175)
[ERROR] at com.carma.swagger.doclet.parser.JaxRsAnnotationParser.run(JaxRsAnnotationParser.java:89)
[ERROR] at com.carma.swagger.doclet.ServiceDoclet.start(ServiceDoclet.java:20)
[ERROR] at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] at java.lang.reflect.Method.invoke(Method.java:483)
[ERROR] at com.sun.tools.javadoc.DocletInvoker.invoke(DocletInvoker.java:310)
[ERROR] at com.sun.tools.javadoc.DocletInvoker.start(DocletInvoker.java:189)
[ERROR] at com.sun.tools.javadoc.Start.parseAndExecute(Start.java:366)
[ERROR] at com.sun.tools.javadoc.Start.begin(Start.java:219)
[ERROR] at com.sun.tools.javadoc.Start.begin(Start.java:205)
[ERROR] at com.sun.tools.javadoc.Main.execute(Main.java:64)
[ERROR] at com.sun.tools.javadoc.Main.main(Main.java:54)
[ERROR] Caused by: java.lang.IllegalArgumentException: duplicate key: Map
[ERROR] at com.google.common.base.Preconditions.checkArgument(Preconditions.java:119)
[ERROR] at com.google.common.collect.RegularImmutableMap.<init>(RegularImmutableMap.java:67)
[ERROR] at com.google.common.collect.ImmutableMap$Builder.fromEntryList(ImmutableMap.java:249)
[ERROR] at com.google.common.collect.ImmutableMap$Builder.build(ImmutableMap.java:235)
[ERROR] at com.google.common.collect.Maps.uniqueIndex(Maps.java:1168)
[ERROR] at com.google.common.collect.Maps.uniqueIndex(Maps.java:1142)
[ERROR] at com.carma.swagger.doclet.parser.CrossClassApiParser.addApiModels(CrossClassApiParser.java:215)

**Comment by shifu** [comment]
@conorroche I was on doclet 1.0.1 and java 8 1.8.0_25-b17 on Mac yosemite.

java version "1.8.0_25"
Java(TM) SE Runtime Environment (build 1.8.0_25-b17)
Java HotSpot(TM) 64-Bit Server VM (build 25.25-b02, mixed mode)

**Comment by conorroche** [comment]
@shifu does it happen for you with version 1.0.3 of the doclet?

**Comment by conorroche** [comment]
closing as cannot reproduce on 1.0.3+ will reopen if further details are provided

**Comment by shifu** [comment]
Confirmed it's fixed in 1.0.3 and reproduced on 1.0.1
